### PR TITLE
Parser Updates

### PR DESCRIPTION
**Pull request by tbreimer14**

### CODE REVIEW

File: parser/src/core.rs
Comment (CalebLItalien) [inline parser/src/core.rs:1]: Should we leave in some of these core functionalities included? If not, maybe we should just leave them out and let the user define them as they need them. If we go that route, we'd just include new and parse.

Comment (CalebLItalien) [inline parser/src/core.rs:1]: Lets remove documentation that documents panics. Ideally we should not include the usage of panics anywhere.

Comment (CalebLItalien) [reply parser/src/core.rs:1]: Should we leave in some of these core functionalities included? If not, maybe we should just leave them out and let the user define them as they need them. If we go that route, we'd just include new and parse, maybe parse_router?

**Review by CalebLItalien — COMMENTED** [review]
Everything looks excellent! A few notes before we close this.

File: parser/src/core.rs
Comment (CalebLItalien) [reply parser/src/core.rs:1]: One more thing, combine import statements wherever possible. IE:
```rust
use common::{ 
    error::ErrorType,
    ast::{
        ast_struct::ASTNode, 
        syntax_element::SyntaxElement,
    },
};
use lexer::token::Token;
use crate::core::Parser;
use common::ast::data_type::DataType;
```

Combine the two common imports into one.

Comment (tbreimer14) [reply parser/src/core.rs:1]: Resolved these issues in the latest push.

**Pull request merged**